library/src/com/google/maps/android/kml/AbsKmlRenderer.java

```diff
@@ -212,10 +224,15 @@ protected void addLayerToMap() {
     if (!mGroundOverlayImagesDownloaded) {
       downloadGroundOverlays();
     }
-    if (!mMarkerIconsDownloaded) {
-      downloadMarkerIcons();
-    }
+
     mLayerVisible = true;
+
+    // XXX: We reuse same layer many times with modified data (e.g. with changed containers visibility) so
+    // we should force icon downloading because without this 'onMarkerIconDownloaded()' will not be called
+    // and right marker icons not set.
+    //if (!mMarkerIconsDownloaded) {
```

Comment:
IMHO, yes. Logic is complicated and this commented code give us some chance of understanding.